tests/unit/views/notifications/test_project_crew_notification.py

```diff
@@ -0,0 +1,400 @@
+def given_vetinari_made_ridcully_admin_org(
+    db_session,
+    user_vimes,
+    user_ridcully,
+    project_expo2010,
+    org_ankhmorpork,
+    user_vetinari,
+):
+    ridcully_admin = models.OrganizationMembership(
+        user=user_ridcully, organization=org_ankhmorpork, granted_by=user_vetinari
+    )
+    db_session.add(ridcully_admin)
+    db_session.commit()
+    return user_ridcully
+
+
+@given(
+    parsers.parse(
+        "Ridcully is an existing crew member of the Ankh-Morpork 2010 project with role {role}"
+    ),
+    target_fixture='ridcully_member',
+)
+def given_ridcully_existing_member(
+    role,
+    db_session,
+    client,
+    login,
+    user_vimes,
+    user_ridcully,
+    project_expo2010,
+    user_vetinari,
+):
+    roles = [_r.strip() for _r in role.split(',')]
+    is_editor = 'editor' in roles
+    is_promoter = 'promoter' in roles
+    is_usher = 'usher' in roles
+    existing_ridcully_member = models.ProjectCrewMembership(
+        parent=project_expo2010,
+        user=user_ridcully,
+        is_editor=is_editor,
+        is_promoter=is_promoter,
+        is_usher=is_usher,
+        granted_by=user_vetinari,
+    )
+    db_session.add(existing_ridcully_member)
+    db_session.commit()
+    return existing_ridcully_member
+
+
+@when(
+    "Vetinari removes Ridcully from the Ankh-Morpork 2010 project crew",
+    target_fixture='ridcully_member',
+)
+def when_ridcully_removed(
+    db_session,
+    user_vetinari,
+    user_vimes,
+    user_ridcully,
+    project_expo2010,
+    ridcully_member,
+):
+    ridcully_member.revoke(actor=user_vetinari)
+    db_session.add(ridcully_member)
+    db_session.commit()
+    return ridcully_member
+
+
+@then(parsers.parse("{user} is notified of the removal with {notification_string}"))
+def then_user_notification_removal(
+    user,
+    notification_string,
+    user_vimes,
+    user_ridcully,
+    user_vetinari,
+    project_expo2010,
+    vimes_promoter,
+    ridcully_member,
+):
+    user_dict = {
+        "Ridcully": user_ridcully,
+        "Vimes": user_vimes,
+        "Vetinari": user_vetinari,
+    }
+    preview = models.PreviewNotification(
+        models.ProjectCrewMembershipRevokedNotification,
+        document=ridcully_member.project,
+        fragment=ridcully_member,
+    )
+    user_notification = models.NotificationFor(preview, user_dict[user])
+    view = user_notification.views.render
+    assert (
+        view.activity_template().format(
+            project=ridcully_member.project.joined_title,
+            user=ridcully_member.user.fullname,
+            actor=ridcully_member.revoked_by.fullname,
+        )
+        == notification_string
+    )
+
+
+@when(
+    "Ridcully resigns from the Ankh-Morpork 2010 project crew",
+)
+def when_ridcully_resigns(
+    db_session,
+    user_vetinari,
+    user_vimes,
+    user_ridcully,
+    project_expo2010,
+    ridcully_member,
+):
+    ridcully_member.revoke(user_ridcully)
+    db_session.add(ridcully_member)
```

Comment:
`db.session.add` is not required unless you've constructed a new object as `Class(...)`. It's already in the session if loaded form the db or if it was created using `membership.replace`.